File: LeetCode/LeetCode75/151.py
class Solution:
    def reverseWords(self, s: str) -> str:
        result=""
        words = s.split(' ')

        for a in reversed(words):
            if a=="":
                continue
            else:
                result+=a
                result+=" "

        return result[:-1]


# Test Cases
if __name__ == '__main__':
    sol=Solution()

    # Test 1
    s="a good   example"
    print("Test 1:", sol.reverseWords(s)) 

File: LeetCode/LeetCode75/151.cpp
#include <iostream>
#include <string>

using namespace std;

class Solution {
public:
    string reverseWords(string s) {
        string result = "";
        string word = "";
        for (char c : s) {
            if (c == ' ' && word == "") {
                continue;
            }
            if (c == ' ') {
                result = word + "*" + result;
                word = "";
            } else {
                word = word + c;
            }
        }
        if (word != "")
            result = word + "*" + result;
        result = result.erase(result.size() - 1);
        return result;
    }
};

int main() {
    Solution s;
    string str = "the sky is blue";
    cout << s.reverseWords(str) << endl;

    str = "       the sky        is blue       ";
    cout << s.reverseWords(str) << endl;
    return 0;
}